src/Drupal/Driver/Fields/Drupal7/AbstractHandler.php

```diff
@@ -22,8 +26,8 @@
    * @param $field_name
    * @return mixed
    */
-  public function __construct($entity_type, $field_name) {
+  public function __construct($entity_type, $field_name, $language) {
     $this->field_info = field_info_field($field_name);
+    $this->language = field_is_translatable('node', $this->field_info) ? $language : LANGUAGE_NONE;
```

Comment:
Yes, comments is where I got my mustard from. I can check out the entity_language if you like and refactor the update.